server/poll/poll.go

```diff
@@ -190,8 +207,19 @@ func (p *Poll) Copy() *Poll {
 	return p2
 }
 
-// EncodeToByte returns a VotedAnswerResponse as a byte array
-func (v *VotedAnswerResponse) EncodeToByte() []byte {
+// WriteTo write Metadata to the io.Writer given as an argument
+func (v *Metadata) WriteTo(w io.Writer) (int64, error) {
 	b, _ := json.Marshal(v)
-	return b
+	i, err := w.Write(b)
+	return int64(i), err
+}
```

Comment:
You can use [`json.Encoder`](https://golang.org/pkg/encoding/json/#NewEncoder) instead:
```suggestion
// WriteTo write Metadata to the io.Writer given as an argument
func (m *Metadata) WriteTo(w io.Writer) error {
	return json.NewEncoder(w).Encode(v)
}
```

Reply:
It looks smart way. But govet check if method `WriteTo(io.Writer)` have the signature like `WriteTo(io.Writer) (int64, error)`.

```
/usr/local/bin/go vet ./...
# github.com/matterpoll/matterpoll/server/poll
server/poll/poll.go:206:20: method WriteTo(w io.Writer) error should have signature WriteTo(io.Writer) (int64, error)
```

`json.NewEncoder(w).Encode(v)` returns only `error` object as return type, so it can't satisfy the rule for signature. We would have some way to fix this.

1. Change method name to avoid govet check (e.g.: `WriterToWriter`)
2. Adopt the existing implementation
3. Revert to `EncodeToByte`, and remove `WriteTo`

I think that **no.3 is the best** for us, because we don't need the size of data to be written. How does this sounds?

Reply:
To be fair, my comment was very nit picky in the first place. So, I'm 0/5 on this.

What about dropping this method and just use `json.Encoder` inside `api.go` in https://github.com/matterpoll/matterpoll/pull/234/files#diff-4444f0caa92e553b5c018c908e90e5e6R528?